### PR TITLE
uint8, int32 and int64 support for 'gather' in paddle backend

### LINKED ISSUE #12592: gather

### PR DESCRIPTION
**Pull request by czirabence**
Close #12592

### CODE REVIEW

**Comment by czirabence** [comment]
> I don't really follow what you've done here, can you explain to me your steps?

This is a correction for my previous pull request #12950, which added gather to paddle backend. Paddle.gather supports dtypes of uint8, int32, int64, float32 and float64 on cpu, according to the [paddlepaddle reference](ttps://www.paddlepaddle.org.cn/documentation/docs/en/2.2/api/paddle/gather_en.html#gather). However, when functional.ivy.gather with paddle backend was given uint8 type array inputs, the following error was raised: ivy.utils.exceptions.IvyBackendException: paddle: gather: (NotFound) The kernel with key (CPU, NCHW, uint8) of kernel `slice` is not registered. To overcome this, 4 list comprehensions inside ivy.functional.bakcends.paddle.gather were replaced with paddle.split. I also removed int32 and int64 from unsupported_devices_and_dtypes, because they are supported.

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Good work
the correct usage would be `paddle.unbind` instead of `split`, while both of them return the correct result, the shape is different because `split` keeps the axis that you split on while `unbind` removes that axis which is the desired behavior, and since unbind doesn't support a lot of dtypes, you can use `ivy.unstack` under `ivy.ArrayMode(False)`
btw try to extend to all dtypes except bfloat and uint16

**Comment by czirabence** [comment]
Thank you,
Paddle functions were replaced with ivy functions in the second commit, and data casting for unsupported types was added, which I copied from backends.paddle.unstack.

File: ivy/functional/backends/paddle/general.py [hunk 1]
```diff
@@ -160,27 +144,50 @@ def gather(
...
+                    )
+                result = ivy.concat(result, axis=0)
+            new_shape = (
+                params1.shape[:axis] + indices.shape[batch_dims:] + params1.shape[axis + 1 :]
+            )
+            return ivy.reshape(result, shape=new_shape)
+        
+    
     axis = axis % paddle.Tensor.ndimension(params)
     batch_dims = batch_dims % paddle.Tensor.ndimension(params)
```
Comment (MahmoudAshraf97) [inline ivy/functional/backends/paddle/general.py:173]:
```suggestion
    axis = axis % params.ndim
    batch_dims = batch_dims % params.ndim
```

**Review by MahmoudAshraf97 — APPROVED** [review]
Great work overall

**Pull request merged**